codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/UnionGenerator.kt

```diff
@@ -26,6 +26,12 @@ fun CodegenMode.renderUnknownVariant() = when (this) {
     is CodegenMode.Client -> true
 }
 
+const val UnknownVariantError = """
+Cannot serialize  `${UnionGenerator.UnknownVariantName}::Unknown` for the request.
```

Comment:
I think this needs to be a function so that the union name can be passed in. As it is, this will format to `Unknown::Unknown`.